Sources/Plasma/FeatureLib/pfPython/pyImageGlue.cpp

```diff
@@ -165,21 +165,34 @@ PYTHON_METHOD_DEFINITION(ptImage, saveAsJPEG, args)
 
 PYTHON_METHOD_DEFINITION(ptImage, saveAsPNG, args)
 {
-    PyObject* filenameObj;
-    if (!PyArg_ParseTuple(args, "O", &filenameObj))
+    PyObject* filenameObj, *optionalText;
+    if (!PyArg_ParseTuple(args, "O|O", &filenameObj, &optionalText))
     {
-        PyErr_SetString(PyExc_TypeError, "saveAsPNG expects a string");
+        PyErr_SetString(PyExc_TypeError, "saveAsPNG expects a string, and an optional dict of key-value pairs");
         PYTHON_RETURN_ERROR;
     }
 
     if (PyString_CheckEx(filenameObj))
     {
-        self->fThis->SaveAsPNG(PyString_AsStringEx(filenameObj));
+        if (PyDict_Check(optionalText)) {
+            std::multimap<ST::string, ST::string> textFields;
+            PyObject *key, *value;
+            Py_ssize_t pos = 0;
+            while (PyDict_Next(optionalText, &pos, &key, &value)) {
+                if (PyString_CheckEx(key) && PyString_CheckEx(value)) {
+                    textFields.emplace(PyString_AsStringEx(key), PyString_AsStringEx(value));
+                }
```

Comment:
Want to check for errors?  Otherwise, this will just silently ignore any entries in the dict that don't have the correct types.

Reply:
We could auto-convert to string values.  I don't think failing with an error would be preferable to skipping bad values and continuing to complete the function's main purpose.  Reporting a warning to the log would be nice as a compromise.  Can we do that from inside the glue functions?

I think the better option would be to use the string representation of the values provided.